### PR TITLE
[CALCITE-3552] Support MySQL ExtractValue function

### PR DESCRIPTION
**Pull request by ritesh-kapoor**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/fun/SqlExtractValueFunction.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.apache.calcite.sql.type.SqlOperandCountRanges;
+import org.apache.calcite.sql.type.SqlOperandTypeChecker;
+import org.apache.calcite.sql.type.SqlTypeTransforms;
+import org.apache.calcite.sql.validate.SqlValidator;
+
+/**
+ * The <code>EXTRACTVALUE</code> function.
+ */
+public class SqlExtractValueFunction extends SqlFunction {
+
```
Comment (danny0405) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlExtractValueFunction.java]: This class is not that necessary, use SqlFunction directly is okey. We only need new class when we want to customize the operands/return type inference.

Comment (chunweilei) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlExtractValueFunction.java]: +1

Comment (ritesh-kapoor) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlExtractValueFunction.java]: Done :)

File: core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java [hunk 2]
```diff
@@ -172,6 +172,13 @@ private SqlLibraryOperators() {
   @LibraryOperator(libraries = {MYSQL, ORACLE})
   public static final SqlFunction REGEXP_REPLACE = new SqlRegexpReplaceFunction();
 
+  @LibraryOperator(libraries = {MYSQL})
+  public static final SqlFunction EXTRACT_VALUE = new SqlFunction(
+      "EXTRACTVALUE", SqlKind.OTHER_FUNCTION,
+      ReturnTypes.cascade(ReturnTypes.VARCHAR_2000, SqlTypeTransforms.FORCE_NULLABLE),
+      null, OperandTypes.STRING_STRING, SqlFunctionCategory.SYSTEM);
+
+
```
Comment (chunweilei) [inline core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java]: Oracle also has this function[1]. Is it the same?

[1] https://docs.oracle.com/cd/B19306_01/server.102/b14200/functions052.htm

Comment (ritesh-kapoor) [reply core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java]: Reading the docs, seems like there is difference in behaviour of MySql ExtractValue and Oracle ExtractValue.

**Pull request merged**